src/features/transfer/TransferTokenForm.tsx

```diff
@@ -218,17 +217,18 @@ function ChainSelectSection({ isReview }: { isReview: boolean }) {
   }, [values.destination, warpCore]);
 
   const { originToken, destinationToken } = useMemo(() => {
-    const originToken = getTokenByIndex(warpCore, values.tokenIndex);
+    const originToken = getTokenByKey(warpCore, values.tokenKey);
     if (!originToken) return { originToken: undefined, destinationToken: undefined };
     const destinationToken = originToken.getConnectionForChain(values.destination)?.token;
     return { originToken, destinationToken };
-  }, [values.tokenIndex, values.destination, warpCore]);
+  }, [values.tokenKey, values.destination, warpCore]);
 
   const setTokenOnChainChange = (origin: string, destination: string) => {
-    const tokenIndex = getTokenIndexFromChains(warpCore, null, origin, destination);
-    const token = getTokenByIndex(warpCore, tokenIndex);
+    // Returns token key if only one route exists, undefined if 0 or multiple routes
+    const tokenKey = getTokenKeyFromChains(warpCore, null, origin, destination);
+    const token = tokenKey ? getTokenByKey(warpCore, tokenKey) : undefined;
     updateQueryParam(WARP_QUERY_PARAMS.TOKEN, token?.symbol);
```

Comment:
With the new token keys, is it still OK to have the symbol as part of the query params? If a user was to share this URL, would it open the app in the same state?